packages/graphql/src/translate/create-connect-or-create-and-params.ts

```diff
@@ -88,7 +96,7 @@ function createConnectOrCreatePartialStatement({
     relationField: RelationField;
     refNode: Node;
     context: Context;
-}): CypherBuilder.Query {
```

Comment:
**question:** What was wrong with "Query" out of interest?

Reply:
Query is not really accurate, as in Cypher terms what we are building is a `Clause`.

I also think is it a much better name, as it avoid ambigüity with the several `query` things that we have in the project

Reply:
Whilst this is true, I've never once said to myself "time to open up Neo4j Browser and write a quick Cypher clause!" - is it not worthwhile using phrasing that we are all familiar with in the Cypher builder?

Reply:
They are not the same, a query is a full input you throw to Neo4j, a Clause is one type of statement supported by Cypher.

So, technically speaking, all clauses can be queries, but the inverse is not necessarily true. Talking about CypherBuilder, a query would be the whole tree (basically, what happens when you call `.build`. But each AST node of the tree is either a Clause or any other of the elements that compose a Cypher Query. So, I think it is way more accurate to say that `Cypher.Match` is a clause than a query. 

It is true, though, that the definition of clause is stretched a bit in the CypherBuilder. `CompositeClause` (`CypherBuilder.concat`) is no longer a clause but can be treated as one, as for simplicity's sake it is classed as such.

We could have 2 levels of semantics here (Clause and Query) or a type named Query that works as some kind of syntax sugar, but tbh considering how overused and meaningless the word query is already in the codebase, it feels redundant. Because of that, I decided to only name `query` the actual string result of the CypherBuilder after `.build` is executed

Reply:
Perhaps we shouldn't be talking in terms of clauses and queries at all for the "top-level" `CypherBuilder` object given that it isn't a query until `.build()` is called?

What if we had a `CypherBuilder.Instance` which contains the `.build()` method which then returns a query? I feel like perhaps we trying to apply an abstraction to something where it doesn't really fit.